meta-resin-common/recipes-support/resin-hostname/resin-hostname/resin-hostname

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+set -e
+
+. /usr/sbin/resin-vars
+
+if [ -z "$NEW_HOSTNAME" ]; then
+    NEW_HOSTNAME=$(jq -r '.uuid //empty' $CONFIG_PATH)
```

Comment:
@petrosagg if there is no HOSTNAME defined in config.json, this script will use UUID as hostname.